### ISSUE TITLE
Deploy using powershell script or Bash script?

### ISSUE THREAD

**Issue by asadsahi** [mkloubert/vs-deploy#17]
@mkloubert trying to run a powershell script with deploy extension, but this is what I get:

![image](https://cloud.githubusercontent.com/assets/3470892/21542400/4f7aa52a-cdb5-11e6-92ff-222f88103daa.png)

And a deploying... icon at the bottom:
![image](https://cloud.githubusercontent.com/assets/3470892/21542417/73072022-cdb5-11e6-8812-9fd39a2fd61d.png)

Second issue I found is that while it is deploying and error occurs, then if you change the settings.json file and save another Deploy button appears, so this just goes on and on an on as shown below:

![image](https://cloud.githubusercontent.com/assets/3470892/21542446/a6f17572-cdb5-11e6-990e-0d041d17db33.png)

As a side note I am also having same problem when trying to deploy targetting a bash script.

**Comment by mkloubert** [comment]
For running a PowerShell script, I have the following solution:

Create a wrapper batch script inside the same directory where your PowerShell script is stored (`deploy.azure.cmd` e.g.) and save the following content:

```batch
@ECHO OFF
CLS

powershell -file "./deploy.azure.ps1"

EXIT
``` 

The target configuration should look like that:

```json
{
    "deploy": {
        "targets": [
            {
                "type": "app",
                "app": "deploy.azure.cmd"
            }
        ]
    }
}
```

Unfortunately, I could not reproduce the "deploy button" problem. **Maybe it disapears since the new version 4.0.1.**

**Comment by asadsahi** [comment]
Trying to deploy using Buttons. Here is my configuration:
![image](https://cloud.githubusercontent.com/assets/3470892/21562393/8d410fd2-ce70-11e6-9dda-c6d6666a9039.png)

VS code hangs when I press putton. And after couple of minutes, this is what I get in console.

![image](https://cloud.githubusercontent.com/assets/3470892/21562409/a9c34ff8-ce70-11e6-9968-3f8658c593f3.png)

Also, button disappears from the bottom as well and reinstated after restarting VS code.

**Comment by mkloubert** [comment]
Oh, the list of arguments (that also contain the files) is too long.

**Since version 4.3.0** I have added a [submitFileList](https://github.com/mkloubert/vs-deploy/wiki/target_app) property, that indicates if files list should be submitted to an app or not (this is `(true)` by default).

You should try the following:

```json
{
    "deploy": {
        "targets": [
            {
                "type": "app",
                "app": "deploy.azure.cmd",

                "submitFileList": false
            }
        ]
    }
}
```

But this mean, that your script / app has to get the list of files by its own.

**Comment by asadsahi** [comment]
@mkloubert It is still not as streamlined as it should be. With this option it starts deploying using script provided in a new command window, but vscode just hangs and in the console of vscode it does something with all the files (including node_modules, that means thousands of files)

![image](https://cloud.githubusercontent.com/assets/3470892/21572198/78cfa37a-cece-11e6-8712-b16c12406927.png)

**Comment by asadsahi** [comment]
I would also comment if we can have native support for ms deploy as that is quite common for asp.net applications rather going through few other options to somehow make it work.

**Comment by mkloubert** [comment]
The problem is, that I cannot get information what file has been finished, because the "app" target simply sends the list of files to the app (if not disabled) and waits until the execution has been finished (this is why the execution "hangs").

**But I have added support for Web Deploy since Version 4.6.0.**

You can use it by defining an ["after deploy" task](https://github.com/mkloubert/vs-deploy/wiki/targetoperations#webdeploy-), e.g.

```json
{
    "deploy": {
        "targets": [
            {
                "type": "test",
                "name": "This is a test",

                "deployed": [
                    {
                        "type": "webdeploy",

                        "exec": "C:/Program Files/IIS/Microsoft Web Deploy V3/msdeploy.exe",
                        "dir": "C:/path/to/working/directory",

                        "verb": "sync",
                        "source": "contentPath=c:\\sourcedir",
                        "dest": "contentPath=c:\\newdir",
                        "whatif": true
                    }
                ]
            }
        ]
    }
}
```

This is same as the following command line execution:

```batch
"C:\Program Files\IIS\Microsoft Web Deploy V3\msdeploy.exe" -verb:sync -source:contentPath=c:\sourcedir -dest:contentPath=c:\newdir -whatif
```

**Comment by asadsahi** [comment]
Thanks @mkloubert will give it a try.

**Comment by mkloubert** [comment]
@asadsahi I think I could reproduce the **problem with the non disapearing buttons on error**.

**The problem should be fixed since version 4.19.1.**

**Comment by asadsahi** [comment]
Ohh okay, thanks. :)
